src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1678,35 +1752,27 @@ public void ProcessFileChange(string projectSystemFilePath, string workspaceFile
                                 return;
                             }
 
-                            if (fileInfoProvider == null)
-                            {
-                                var textLoader = new FileTextLoader(projectSystemFilePath, defaultEncoding: null);
-                                _documentTextLoaderChangedAction(w, documentId, textLoader);
-                            }
-                            else
-                            {
-                                // we do not expect JTF to be used around this code path. and contract of fileInfoProvider is it being real free-threaded
-                                // meaning it can't use JTF to go back to UI thread.
-                                // so, it is okay for us to call regular ".Result" on a task here.
-                                var fileInfo = fileInfoProvider.GetDynamicFileInfoAsync(
-                                    _project.Id, _project._filePath, projectSystemFilePath, CancellationToken.None).WaitAndGetResult_CanCallOnBackground(CancellationToken.None);
-
-                                // Right now we're only supporting dynamic files as actual source files, so it's OK to call GetDocument here
-                                var document = w.CurrentSolution.GetRequiredDocument(documentId);
-
-                                var documentInfo = DocumentInfo.Create(
-                                    document.Id,
-                                    document.Name,
-                                    document.Folders,
-                                    document.SourceCodeKind,
-                                    loader: fileInfo.TextLoader,
-                                    document.FilePath,
-                                    document.State.Attributes.IsGenerated,
-                                    document.State.Attributes.DesignTimeOnly,
-                                    documentServiceProvider: fileInfo.DocumentServiceProvider);
-
-                                w.OnDocumentReloaded(documentInfo);
-                            }
+                            // we do not expect JTF to be used around this code path. and contract of fileInfoProvider is it being real free-threaded
+                            // meaning it can't use JTF to go back to UI thread.
+                            // so, it is okay for us to call regular ".Result" on a task here.
```

Comment:
In devenv, it is never OK to call into external code without JTF. Each team implements their own code according to the JTF rules with the understanding that they will interoperate correctly. The use of `.Result` violates the cross-team contract and can (and has in cases) lead to deadlock.

Reply:
This is unchanged in this PR.

Reply:
Bad luck the diff didn't hide it from me 🤣

Reply:
Oh, I wasn't expecting the diff to hide it; the existing behavior is moot from the point of this PR though. :smile: